apps/rxjs/race-condition/src/app/app.component.ts

```diff
@@ -12,27 +7,26 @@ import { TopicService, TopicType } from './topic.service';
 @Component({
   standalone: true,
   selector: 'app-root',
-  template: ` <button (click)="openTopicModal()">Open Topic</button> `,
+  template: `
+    <button (click)="openTopicModal()">Open Topic</button>
+  `,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AppComponent implements OnInit {
+export class AppComponent {
   title = 'rxjs-race-condition';
   dialog = inject(MatDialog);
   topicService = inject(TopicService);
-  topics: TopicType[] = [];
 
-  ngOnInit(): void {
+  openTopicModal() {
     this.topicService
       .fakeGetHttpTopic()
       .pipe(take(1))
-      .subscribe((topics) => (this.topics = topics));
-  }
-
-  openTopicModal() {
-    this.dialog.open(TopicModalComponent, {
-      data: {
-        topics: this.topics,
-      },
-    });
+      .subscribe((topics: TopicType[]) => {
```

Comment:
why not but it means that you will need to make a http call each time you open the modal. But maybe you want fresh data each time. But in that case, it's maybe better to make the call inside the modal directly. because if the call takes 10s, the modal will not open and the UX is not good for the user.

Reply:
Yeah, you're right. I did not notice it. :) Thanks for validation,